pipeline/run_pipeline.py

```diff
@@ -480,150 +492,113 @@ def run_pipeline(model_path, refusal_direction=None, activation_prompts=False):
-            cfg,
-            "ablation",
-            dataset_name,
-            eval_methodologies=cfg.jailbreak_eval_methodologies,
-        )
-        evaluate_completions_and_save_results_for_dataset(
-            cfg,
-            "actadd",
-            dataset_name,
-            eval_methodologies=cfg.jailbreak_eval_methodologies,
-        )
-
-    # 4a. Generate and save completions on harmless evaluation dataset
-    harmless_test = random.sample(
-        load_dataset_split(harmtype="harmless", split="test"), cfg.n_test
+    harmful_test = random.sample(
+        load_dataset(harmful_dataset_name), cfg.n_test
     )
-
     generate_and_save_completions_for_dataset(
         cfg,
         model_base,
         baseline_fwd_pre_hooks,
         baseline_fwd_hooks,
         "baseline",
-        "harmless",
-        dataset=harmless_test,
+        harmful_dataset_name,
+        temperature,
+        dataset=harmful_test,
     )
 
-    actadd_refusal_pre_hooks, actadd_refusal_hooks = (
-        [
-            (
-                model_base.model_block_modules[layer],
-                get_activation_addition_input_pre_hook(
-                    vector=refusal_direction, coeff=+1.0
-                ),
-            )
-        ],
-        [],
+    harmless_test = random.sample(
+        load_dataset_split(harmtype="harmless", split="test"),
+        cfg.n_test,
     )
     generate_and_save_completions_for_dataset(
         cfg,
         model_base,
-        actadd_refusal_pre_hooks,
-        actadd_refusal_hooks,
-        "actadd",
+        baseline_fwd_pre_hooks,
+        baseline_fwd_hooks,
+        "baseline",
         "harmless",
+        temperature,
         dataset=harmless_test,
     )
 
-    # 4b. Evaluate completions and save results on harmless evaluation dataset
-    evaluate_completions_and_save_results_for_dataset(
-        cfg, "baseline", "harmless", eval_methodologies=cfg.refusal_eval_methodologies
-    )
-    evaluate_completions_and_save_results_for_dataset(
-        cfg, "actadd", "harmless", eval_methodologies=cfg.refusal_eval_methodologies
-    )
+    for layer in range(model_base.model.config.num_hidden_layers):
+        ablation_fwd_pre_hooks, ablation_fwd_hooks = get_all_direction_ablation_hooks(
+            model_base, refusal_direction
+        )
 
-    # 5. Evaluate loss on harmless datasets
-    evaluate_loss_for_datasets(
-        cfg, model_base, baseline_fwd_pre_hooks, baseline_fwd_hooks, "baseline"
-    )
-    evaluate_loss_for_datasets(
-        cfg, model_base, ablation_fwd_pre_hooks, ablation_fwd_hooks, "ablation"
-    )
-    evaluate_loss_for_datasets(
-        cfg, model_base, actadd_fwd_pre_hooks, actadd_fwd_hooks, "actadd"
-    )
+        actadd_fwd_pre_hooks, actadd_fwd_hooks = (
+            [
+                (
+                    model_base.model_block_modules[layer],
+                    get_activation_addition_input_pre_hook(
+                        vector=refusal_direction, coeff=-1.0
```

Comment:
Can you confirm whether `coeff=-1.0` is correct or not?

Reply:
yes, 
        vector = vector.to(activation)
        activation += coeff * vector
        if isinstance(input, tuple):
            return (activation, *input[1:])
        else:
            return activation
            
    so they are doing 2 things on harmful dataset, 
    1)first ablating using the standard method ( x- (x.r)x)
    2) they are just directly subtracting the refusal direction ( x- r )